extension/agenthealth/config.go

```diff
@@ -12,6 +12,7 @@ import (
 type Config struct {
 	IsUsageDataEnabled bool              `mapstructure:"is_usage_data_enabled"`
 	Stats              agent.StatsConfig `mapstructure:"stats"`
+	StatusCodeOnly     *bool             `mapstructure:"is_status_code_only,omitempty"`
```

Comment:
Agreed with mitali. Instead of a pointer, can we treat the default value as false?

Additionally, can you explain a bit more what this field is used for? It seems like a way to set the agent health extension such that it turns the other things off. I think that may be confusing because you could set `IsUsageDataEnabled` to true and also set `StatusCodeOnly` to true, so it's not clear what it would actually do (without looking at the implementation).

Reply:
the pointer is so that we can emit it if it is empty(nil)